lib/src/iterable_extensions.dart

```diff
@@ -353,6 +354,12 @@ extension IterableExtension<T> on Iterable<T> {
     return null;
   }
 
+  /// Associates the elements in [this] by the value returned by [key].
+  ///
+  /// Returns a map from keys computed by [key] to the last value for which [key]
+  /// returns that key.
```

Comment:
I'm not exactly sure what you mean - do you mean the order in the resulting `Map`? In which case I'd argue that it's implied by the signature, using a generic `Map` rather than a more specific data structure.